src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -312,7 +322,47 @@ public bool ForEach(KeyValuePair<string, object> activityTag)
                 }
                 else if (activityTag.Value != null)
                 {
-                    ProcessJaegerTag(ref this, activityTag.Key, activityTag.ToJaegerTag());
+                    var key = activityTag.Key;
+                    var jaegerTag = activityTag.ToJaegerTag();
+                    if (jaegerTag.VStr != null)
+                    {
+                        PeerServiceResolver.InspectTag(ref this, key, jaegerTag.VStr);
+
+                        if (key == SpanAttributeConstants.StatusCodeKey)
+                        {
+                            StatusCode? statusCode = StatusHelper.GetStatusCodeForTagValue(jaegerTag.VStr);
+                            if (statusCode == Trace.StatusCode.Error)
+                            {
+                                // Error flag: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#error-flag
+                                this.StatusCode = statusCode;
+                                return true;
+                            }
+                            else if (!statusCode.HasValue || statusCode == Trace.StatusCode.Unset)
+                            {
+                                // Unset Status is not sent: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#status
+                                return true;
+                            }
+
+                            this.StatusCode = statusCode;
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3073/files#diff-6d1626406ea3dabab7357969eb1e8401d78114acaf5e916f6a85f300b23afe96R331-R348 - This whole block is doing unnecessary checks. It should just check if this tag is storing statuscode, and if yes, store it to this.StatusCode. We dont need of the other checks, as it is handled elsewhere.

Reply:
It will provide early termination for the tags and perform normalization. I can merge to normalization logic onto the  ToJaegerSpan if we think that is a better way to go.

Reply:
I am not sure what is the early termination benefit here. Its doing some checks, which are repeated when we set jaeger tags again.

The following should be enough
```
if (key == SpanAttributeConstants.StatusCodeKey)
{
StatusCode? statusCode = StatusHelper.GetStatusCodeForTagValue(jaegerTag.VStr);
this.StatusCode = statusCode;
return true;
}
```